warehouse/templates/manage/token.html

```diff
@@ -133,23 +135,42 @@ <h2>{% trans %}Add another token{% endtrans %}</h2>
         <span class="form-group__label">{% trans %}Permissions{% endtrans %}</span>
         <p class="form-group__text">{% trans %}Upload packages{% endtrans %}</p>
       </div>
-      <div class="form-group">
-        <label for="token_scope" class="form-group__label">
+      <div class="form-group token-scopes" data-controller="token-scopes">
+        <label for="token_scopes" class="form-group__label">
           {% trans %}Scope{% endtrans %}
-          {% if create_macaroon_form.token_scope.flags.required %}
+          {% if create_macaroon_form.token_scopes.flags.required %}
           <span class="form-group__required">{% trans %}(required){% endtrans %}</span>
           {% endif %}
         </label>
-        <select name="token_scope" id="token_scope" class="form-group__input" aria-describedby="scope-errors" autocomplete="off">
-          <option disabled {{ "" if create_macaroon_form.token_scope.data else "selected" }} value="scope:unspecified">{% trans %}Select scope...{% endtrans %}</option>
-          <option value="scope:user">{% trans %}Entire account (all projects){% endtrans %}</option>
+        {% set token_scopes = create_macaroon_form.token_scopes.data %}
+        {% set all_projects = not token_scopes or token_scopes == ["scope:user"] %}
+        <div>
+          <input type="radio" name="token_scopes" value="scope:user" id="scope:user" data-action="token-scopes#user" {{ "checked" if all_projects }}>
+          <label for="scope:user">{% trans %}Entire account (all projects){% endtrans %}</label>
+        </div>
+        <div>
+          <input type="radio" name="token_scopes" value="scope:by_project" id="scope:by_project" data-action="token-scopes#project" {{ "checked" if not all_projects }}>
+          <label for="scope:by_project">{% trans %}By project{% endtrans %}</label>
```

Comment:
Could you explain the distinction between `scope:project` and `scope:by_project` here?

If I'm understanding correctly `scope:project` gets used for a single project selection and `scope:by_project` gets used for a multi-selection, but IMO it might make sense to treat either `scope:` as a list and just standardize on one.

Reply:
> If I'm understanding correctly scope:project gets used for a single project selection and scope:by_project gets used for a multi-selection

Yes. Although I'm not sure we actually `scope:by_project` at all 🤔 

> but IMO it might make sense to treat either scope: as a list and just standardize on one

Not sure what you mean, sorry.

To clarify a bit what's happening, all the token related inputs in the form use `token_scopes` as their name. The backend's form validation reads the list of values and produces the list as required.

Reply:
> Although I'm not sure we actually need scope:by_project at all 🤔

Given it's a radio button both inputs need to have the same name. We could change the name of the per project token inputs, but we still have to pass the "by_project" value, whatever it may be, and ignore it if it's active in the form validation logic so we wouldn't be gaining a lot.

Reply:
I'm also not following what the distinction is here. I think what @woodruffw is saying is: why can't we use the same scope for both of them, and handle `token_scopes` as being a list with one scope (per-project) or multiple scopes (multi-project)?

Reply:
Right, I think I understand.

The only valid values for tokens scopes are `scope:user` and a list of one or more `scope:project:<PROJECT_NAME>`, i.e. there is no `scope:project` alone without a `:<PROJECT_NAME>` suffix. The form validation ensures the user cannot mix `scope:user` and `scope:project:<PROJECT_NAME>`.

`scope:by_project` is only there to allow the radio buttons to flick via standard HTML, it is explicitly ignored by the form validation. I could really be called anything else and probably should.

We could also not rely on HTML for the radio button flick, that might make things easier to understand but it would add complexity to the Stimulus controller.

Does that make sense or am I still missing something?

Reply:
Gentle ping on this comment, was I still missing something? Would removing `scope:` from `scope:by_project` suffice?

Reply:
Yes, I think dropping the `scope:` prefix is appropriate given that it isn't parsed during form validation.

Reply:
Pshed the changes to https://github.com/Sterbic/warehouse/pull/2. I'll ask for a re-review once @Sterbic merges it.